src/main/java/com/google/codeu/servlets/MapsServlet.java

```diff
@@ -0,0 +1,56 @@
+package com.google.codeu.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Returns Starbucks Locations as an array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}]
+ */
+@WebServlet("/starbucks-data")
+public class MapsServlet extends HttpServlet {
+
+  JsonArray starbucksLocationsArray;
+
+  @Override
+  public void init() {
+    starbucksLocationsArray = new JsonArray();
+    Gson gson = new Gson();
+    Scanner scanner =
+            new Scanner(getServletContext().getResourceAsStream("/WEB-INF/starbucks-data.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      String state = cells[0];
+      double lat = Double.parseDouble(cells[1]);
+      double lng = Double.parseDouble(cells[2]);
+
+      starbucksLocationsArray.add(gson.toJsonTree(new UfoSighting(state, lat, lng)));
```

Comment:
Maybe consider a UfoSighting constructor that takes a String. Then in the constructor Javadoc, write the expected format of the string. This will abstract some of the more complicated code out of this method.

new UfoSighting(line);

/** 
* Some javadoc here.
* @param dataLine something descriptive
UfoSighting(String dataLine) {
  // Do the parsing here.

}